app.js

```diff
@@ -176,6 +180,111 @@ function resetScheduler() {
   });
 }
 
+const serviceAccountKeyFile = "./article-bot-database-2c885470e1c3.json";
+const sheetId = '19cKf8XBNbSXiQmfsVFT3F63__e_61H8w6iosU7hIKyM'
+const tabName = 'Sheet1'
+const range = 'A:C'
+
+async function _getGoogleSheetClient() {
+  const auth = new google.auth.GoogleAuth({
+    keyFile: serviceAccountKeyFile,
+    scopes: ['https://www.googleapis.com/auth/spreadsheets'],
+  });
+  const authClient = await auth.getClient();
+  return google.sheets({
+    version: 'v4',
+    auth: authClient,
+  });
+}
+
+async function _readGoogleSheet(googleSheetClient, sheetId, tabName, range) {
+  const res = await googleSheetClient.spreadsheets.values.get({
+    spreadsheetId: sheetId,
+    range: `${tabName}!${range}`,
+  });
+
+  return res.data.values;
+}
+
+async function _writeGoogleSheet(googleSheetClient, sheetId, tabName, range, data) {
+  await googleSheetClient.spreadsheets.values.append({
+    spreadsheetId: sheetId,
+    range: `${tabName}!${range}`,
+    valueInputOption: 'USER_ENTERED',
+    insertDataOption: 'INSERT_ROWS',
+    resource: {
+      "majorDimension": "ROWS",
+      "values": data
+    },
+  })
+}
+
+async function _updateGoogleSheet(googleSheetClient, sheetId, tabName,rangeupdate,data){
+  await googleSheetClient.spreadsheets.values.update({
+    spreadsheetId: sheetId,
+    range: `${tabName}!${rangeupdate}`,
+    valueInputOption: 'USER_ENTERED',
+    resource: {
+      "values": data
+    },
+  })
+}
+
+async function main_getarticle(str_id) {
+  // Generating google sheet client
+  const googleSheetClient = await _getGoogleSheetClient();
+  var data_onSheet =await _readGoogleSheet(googleSheetClient, sheetId, tabName, range);
+  idx=0
```

Comment:
what is `idx`?. change it to a more meaningful name

Reply:
I used idx as shortform of index while accessing the array data as index is a keyword

Reply:
change it to some meaningful name